.freeCodeCamp/tooling/validate.js

```diff
@@ -112,5 +113,62 @@ export async function validateCurriculum() {
       );
     }
   }
+
+  for (const seed of seeds) {
+    const project = seed.replace('-seed.md', '.md');
+    const projectDirPath = project.replace('.md', '');
+    try {
+      await access(projectDirPath, constants.F_OK);
+    } catch (e) {
+      throw new Error(
+        `Seed "${seed}" does not have an associated root directory`
+      );
+    }
+
+    const mdPath = `${CURRICULUM_PATH}/${project}`;
+    try {
+      const projectTitle = await getProjectTitle(mdPath);
+      if (!projectTitle.projectTopic || !projectTitle.currentProject) {
+        throw new Error(
+          `Seed "${seed}" has a malformed title: '${projectTitle}'`
+        );
+      }
+    } catch (e) {
+      throw new Error(
+        `Seed "${seed}" has a malformed title: '${e.message}'`
+      );
+    }
+
+    const mdFile = await readFile(mdPath, 'utf8');
+    if (!mdFile.trimEnd().endsWith('--fcc-end--')) {
+      throw new Error(`Seed "${seed}" does not end with --fcc-end--`);
+    }
+
+    const lessonNumbers = [...mdFile.matchAll(/\n## (\d+)/g)]
+      .filter(Boolean)
+      .map(([, num]) => Number(num));
+    if (!lessonNumbers.length) {
+      throw new Error(`Seed "${seed}" does not have any lessons`);
+    }
+
+    let expectedLessonNumber = 1;
+
+    for (const lessonNumber of lessonNumbers) {
+      if (lessonNumber !== expectedLessonNumber) {
+        throw new Error(
+          `Seed "${seed}" has a lesson number mismatch. Expected "${expectedLessonNumber}" but got "${lessonNumber}"`
+        );
+      }
+      expectedLessonNumber++;
+    }
```

Comment:
To be clear, I added this only because currently lesson parser (`getLessonFromFile`) technically needs it for seed file. I used the same regex as for lesson, which expects marker for next lesson - `lessonNumber + 1` to be in file, otherwise matched for lesson will be everything from `lessonNumber` to `--fcc-end--`.

Reply:
I have been mulling this over:

1. The lesson parser should be changed to not require congruent lesson numbers[^1]
2. There should be (if there already is not) validation (on development) that the lesson numbers for a project are congruent
3. The seed section can then make use of the same code without the validation (seed should still be sequential, though)

[^1]: This will be a v2 release